### ISSUE TITLE
friendly errors if fusermount is missing / fuse not loaded

### ISSUE THREAD

**Issue by joshuadfranklin** [jszwedko/ec2-metadatafs#3]
Hi,
If a system does not already have fuse installed and loaded, the error messages are hidden unless you run ec2-metadatafs in the foreground with verbose. 

```
ec2-metadatafs -fv /tmp/aws
# doesn't work, says 2016/02/28 13:37:25 forked child with PID 2826

ec2-metadatafs -fv /tmp/aws
#2016/02/28 13:37:12 [DEBUG] mounting at /tmp/aws directed at http://169.254.169.254/latest/ with options: []
#2016/02/28 13:37:12 mount fail: exec: "/bin/fusermount": stat /bin/fusermount: no such file or directory

yum -y install fuse
ec2-metadatafs -fv /tmp/aws
#2016/02/28 13:38:14 [DEBUG] mounting at /tmp/aws directed at http://169.254.169.254/latest/ with options: []
# /bin/fusermount: fuse device not found, try 'modprobe fuse' first
#2016/02/28 13:38:14 mount fail: fusermount exited with code 256

modprobe fuse
ec2-metadatafs -fv /tmp/aws
# works
```

By the way, something like this reverts to a broken state:

```
umount /tmp/aws
umount /sys/fs/fuse/connections/
rmmod fuse
yum -y remove fuse
```

**Comment by jszwedko** [comment]
:+1: thanks for the report, we should definitely fail in a more visible manner.

**Comment by jszwedko** [comment]
@joshuadfranklin I had some time to dig into this this weekend. Unfortunately I wasn't able to address this directly -- the only "robust" way I could find involved mounting and unmounting in the main process before forking which caused `fstab` to hang during operations like `mount -a`.

I added some functionality to help mitigate this issue in the form of sending logs to syslog when daemonized (#7) and blocking in the parent process until the daemonized process notifies (#8) (with a timeout). What do you think about these mitigations? Is there anything else you might like to see?

**Comment by jszwedko** [comment]
Fixed by #7 #8
